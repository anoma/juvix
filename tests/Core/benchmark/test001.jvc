-- tail recursion

def sum' := \x \acc if x = 0 then acc else sum' (x - 1) (x + acc);
def sum := \x sum' x 0;

def writeLn := \x write x >> write "\n";

writeLn (sum 10000) >>
writeLn (sum 100000) >>
writeLn (sum 1000000) >>
writeLn (sum 10000000) >>
writeLn (sum 100000000) >>
writeLn (sum 1000000000)
