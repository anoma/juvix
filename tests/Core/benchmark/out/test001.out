50005000
5000050000
500000500000
50000005000000
5000000050000000
500000000500000000
